tools/slicec-cs/src/decoding.rs

```diff
@@ -468,39 +468,60 @@ fn decode_func_body(type_ref: &TypeRef, namespace: &str, encoding: Encoding) ->
+        // If there's only one parameter, it isn't tagged, and doesn't require a bit-sequence to decode,
+        // We return a simplified lambda function.
+        [param] if !param.is_tagged() && (encoding == Encoding::Slice1 || !param.data_type().is_optional) => {
+            decode_func(param.data_type(), &param.namespace(), encoding)
+        }
 
-    for parameter in get_sorted_members(non_streamed_parameters) {
-        let param_type = parameter.data_type();
-        let namespace = parameter.namespace();
+        // Otherwise we return a full multi-line lambda function for decoding the parameters.
+        _ => {
+            let mut code = CodeBlock::default();
 
-        // For optional value types we have to use the full type as the compiler cannot
-        // disambiguate between null and the actual value type.
-        let param_type_string = match param_type.is_optional && param_type.is_value_type() {
-            true => param_type.incoming_parameter_type_string(&namespace, false),
-            false => "var".to_owned(),
-        };
+            initialize_bit_sequence_reader_for(non_streamed_parameters, &mut code, encoding);
 
-        let param_name = &parameter.parameter_name_with_prefix();
+            for parameter in get_sorted_members(non_streamed_parameters) {
+                let param_type = parameter.data_type();
+                let namespace = parameter.namespace();
 
-        let decode = match parameter.is_tagged() {
-            true => decode_tagged(parameter, &namespace, false, encoding),
-            false => decode_member(parameter, &namespace, encoding),
-        };
+                // For optional value types we have to use the full type as the compiler cannot
+                // disambiguate between null and the actual value type.
+                let param_type_string = match param_type.is_optional && param_type.is_value_type() {
+                    true => param_type.incoming_parameter_type_string(&namespace, false),
+                    false => "var".to_owned(),
+                };
 
-        writeln!(code, "{param_type_string} {param_name} = {decode};")
-    }
+                let param_name = &parameter.parameter_name_with_prefix();
 
-    writeln!(code, "return {};", non_streamed_parameters.to_argument_tuple());
+                let decode = match parameter.is_tagged() {
+                    true => decode_tagged(parameter, &namespace, false, encoding),
+                    false => decode_member(parameter, &namespace, encoding),
+                };
 
-    code
+                writeln!(code, "{param_type_string} {param_name} = {decode};")
+            }
+            writeln!(code, "return {};", non_streamed_parameters.to_argument_tuple());
+
+            let body_content = code.indent();
+            format!(
+                "\
+    (ref SliceDecoder decoder) =>
+    {{
+        {body_content}
+    }}",
+            )
+            .into()
+        }
+    }
```

Comment:
This formatting looks dubious. It should be left indented all the way. You should review the generated code to see that it looks correct.

Reply:
Yes, good catch!
It got auto-indented when I copy/pasted this function, and I didn't see it.